### PR TITLE
Update GPU e2e tests.

### PR DESCRIPTION
**Pull request by rohitagarwal003**
* Use nvidia driver installer from external repo.
    
    That installer decouples itself from COS image version (as long as the
    image version is newer than cos-stable-59-9460-60-0).
    
    A separate commit in the test-infra repo will update the cos version
    used for this test to cos-stable-59-9460-60-0.

* Use cos-stable-59-9460-60-0 and newer installer for GPU node e2e tests.

This is to enable #47388.

This supercedes #47091.

**Release note**:
```release-note
NONE
```

/sig node

### CODE REVIEW

**Comment by mindprince** [comment]
cc - @vishh

File: test/e2e/nvidia-gpus.go [hunk 1]
```diff
@@ -158,10 +159,16 @@ func testNvidiaGPUsOnCOS(f *framework.Framework) {
...
 
 // dsFromManifest reads a .json/yaml file and returns the daemonset in it.
-func dsFromManifest(fileName string) *extensions.DaemonSet {
+func dsFromManifest(url string) *extensions.DaemonSet {
 	var controller extensions.DaemonSet
-	framework.Logf("Parsing ds from %v", fileName)
-	data := generated.ReadOrDie(fileName)
+	framework.Logf("Parsing ds from %v", url)
+
+	response, err := http.Get(url)
```
Comment (vishh) [inline test/e2e/nvidia-gpus.go]: Should we retry a few times just to avoid spurious network issues or github serving issues?

**Comment by vishh** [comment]
Just one comment. Otherwise LGTM

**Comment by vishh** [comment]
@k8s-bot ok to test

File: test/e2e/nvidia-gpus.go (hunk 1 above)
Comment (mindprince) [inline test/e2e/nvidia-gpus.go]: Added retries.

**Comment by mindprince** [comment]
Waiting for https://github.com/kubernetes/test-infra/pull/3050 to get merged.

**Pull request merged**